### PR TITLE
Fixed commit decorator rollbacked on 2xx and 3xx statuses (#304)

### LINKED ISSUE #299: Bug on commit decorator while redirection
The `commit` decorator rollback the session changes when HTTP redirection happens.

### LINKED ISSUE #300: Bug on commit decorator while raise 2xx statuses
The `commit` decorator rollback the session changes when raising 2xx statuses and its children (HTTPSuccess, HTTPNoContent, ...).
https://github.com/Carrene/restfulpy/blob/6313047f6c6cef2ed14dee82225f78bc6dff3b36/restfulpy/orm/__init__.py#L90-L93

Duplicate: #299

### PR DESCRIPTION
**Pull request by ArashFatahzade**
Fixed commit decorator rollbacking database session on status 2xx and 3xx.

### CODE REVIEW

File: restfulpy/tests/test_commit_decorator.py [hunk 1]
```diff
@@ -72,3 +88,31 @@ def test_rollback(self):
         with self.given('Raise exception', verb='ERROR', url='/'):
             assert response.status == 500
 
+    def test_commit_on_raise_http_success(self):
+        with self.given(
+            'Testing the operation of commit decorator on raise 2xx',
+            verb='SUCCESS',
```
Comment (pylover) [inline restfulpy/tests/test_commit_decorator.py]: No need to mention `verb=` and `url=` if you preserve the order. check python's documentation.

File: restfulpy/tests/test_commit_decorator.py [hunk 2]
```diff
@@ -72,3 +88,31 @@ def test_rollback(self):
...
+                 url='/HTTPSuccess'
+            )
+            assert response.json['title'] == 'HTTPSuccess'
+
+    def test_commit_on_raise_http_redirect(self):
+        with self.given(
+            'Testing the operation of commit decorator on raise 3xx',
+            verb='REDIRECT',
+            url='/',
+            form=dict(title='HTTPRedirect')
```
Comment (pylover) [inline restfulpy/tests/test_commit_decorator.py]: Why use the name of an error `HTTPRedirect` in the form parameters?

File: restfulpy/tests/test_commit_decorator.py [hunk 3]
```diff
@@ -72,3 +88,31 @@ def test_rollback(self):
...
+        ):
+            when(
+                'Geting the result of appling commit decorator',
+                 verb='GET',
+                 url='/HTTPRedirect'
```
Comment (pylover) [inline restfulpy/tests/test_commit_decorator.py]: Upper-case URL? really?

File: restfulpy/orm/__init__.py [hunk 4]
```diff
@@ -87,7 +87,12 @@ def wrapper(*args, **kwargs):
             DBSession.commit()
             return result
 
-        except:
+        except Exception as ex:
+            # Actualy 200 <= status <= 399 is not an exception and commit must
+            # be occure.
+            if hasattr(ex, 'status') and '200' <= ex.status < '400':
```
Comment (pylover) [inline restfulpy/orm/__init__.py]: Very implicit, why not type checking?

**Comment by pylover** [comment]
What's your meaning about `Code enhancement` please describe it.

**Comment by pylover** [comment]
#300 and #299 are already closed, why you send PR to close them?

**Comment by ArashFatahzade** [comment]
> #300 and #299 are already closed, why you send PR to close them?

Created a new issue for this bug. #308

> What's your meaning about `Code enhancement` please describe it.

Sorry for my incomplete commit message. I rewrote commit history.

File: restfulpy/tests/test_commit_decorator.py [hunk 5]
```diff
@@ -69,6 +78,35 @@ def test_commit_decorator_and_json_patch(self):
...
+    def test_commit_on_raise_http_redirect(self):
+        with self.given(
+            'Testing the operation of commit decorator on raise 3xx',
+            '/',
+            'POST',
+            form=dict(title='redirect')
+        ):
+            when(
+                'Geting the result of appling commit decorator',
+                 url='/redirect',
```
Comment (pylover) [inline restfulpy/tests/test_commit_decorator.py]: Remove `url=` and `verb=`

**Pull request merged**